feat(calendar): maintain parity with skin's a11y updates

LuLaValva opened the pull request:
<!-- Delete any sections below that are not relevant. -->

- more details on ebay/skin#2019
    - until merged, these changes won't work unless ebayui-core is locally linked with that skin on that branch

## Description
- Add configurable a11y text for "today" and "selected"
- Move a11y info to clipped span in static calendars
- Add classes and move attributes to satisfy new skin selectors

agliga commented on src/components/ebay-calendar/index.marko:
```diff
@@ -104,10 +103,12 @@ $ const rangeEndText = input.a11yRangeEndText ?? "end of range"
-                                                class=disabled && "calendar__cell--disabled"
-                                                aria-label=rangeAriaLabel>
-                                                ${day}
+                                            $ if (isToday) a11yText += ` - ${todayText}`;
```
How is this going to be localized? We also have to consider that in some other languages the day goes before the action. 
I think we should allow this to be fully changeable. We can give them the tools they need to do so however.

LuLaValva replied:
I was thinking about this when I wrote the code, which is why I have hyphens between each label. So the longest label will be something like
```
10 - [selected] - [start of range] - [today]
```
where all of the text in `[]` brackets  is configurable. Is this not enough for localization?

ianmcburnie replied:
Yes to what Luke says. The idea with the hyphens is that this is not a grammatical construct/sentence. Instead we are mimicking ARIA states whereby we control the order in which they are read.

agliga commented on src/components/ebay-date-textbox/date-textbox.stories.js:
```diff
@@ -91,23 +91,16 @@ export default {
-            control: { type: 'text' },
-            description: 'Text to indicate the end of a selected range',
-            table: { defaultValue: { summary: 'end of range' } },
+        buildA11yCellText: {
```
Can we have an example of this in storybook?

agliga reviewed (commented):
The only change id like is the changing of params. The rest we can do post release.

agliga reviewed (commented):
Also should probably document it in date textbox too. I looked if it was but i didnt see it.

LuLaValva reviewed (commented):
Thanks for the review Andrew, I'd like to follow up with additional documentation but these changes should be in soon so for now I think this will suffice

ArtBlue commented on src/components/ebay-date-textbox/examples/localized.marko:
```diff
@@ -0,0 +1,15 @@
+    locale="af"
+    getA11yShowMonthText=(monthName) => `Gaan na ${monthName}`
+    buildA11yCellText=({isSelected, isRangeStart, isInRange, isRangeEnd}) => {
+        let string = ""
```
missing `;`

ArtBlue reviewed (approved):
Approved; just add the missing `;` Nice work!

ianmcburnie commented:
One thing I noticed is `aria-selected` on the `<td>` elements for start and end of range. These ARIA semantics are redundant and will not be announced as user navigates from button to button. 

We should remove this and either:

* A) Add "selected" to clipped text of button
* B) Just let the "start of range" and "end of range" clipped text suffice

<img width="1560" alt="Screen Shot 2023-03-28 at 6 30 56 PM" src="https://user-images.githubusercontent.com/38065/228403667-ced864d8-f82b-4e39-98ba-df2fa0b47e99.png">

LuLaValva commented:
> One thing I noticed is `aria-selected` on the `<td>` elements for start and end of range. These ARIA semantics are redundant and will not be announced as user navigates from button to button.

@ianmcburnie you may still have an older version of this branch cloned locally, I changed that behavior last week

> I also think aria-pressed on the start and end buttons might be a good idea. And on single select use case too.

Ok, I've added add `aria-pressed` to the selected buttons.

agliga commented on src/components/ebay-calendar/index.marko:
```diff
@@ -64,37 +69,33 @@ $ const rangeEndText = input.a11yRangeEndText ?? "end of range"
-                                    <td
-                                        class={
+                                    $ const isDisabled = component.isDisabled(dayISO);
+                                    $ let a11yTexts = [""];
+                                    $ if (isSelected) a11yTexts.push(a11ySelectedText);
+                                    $ if (isRangeStart) a11yTexts.push(a11yRangeStartText);
+                                    $ if (!isRangeStart && !isRangeEnd && isInRange) a11yTexts.push(a11yInRangeText);
+                                    $ if (isRangeEnd) a11yTexts.push(a11yRangeEndText);
```
Can we use a block syntax for this?

```
$ {
let a11yTexts;
if etc
};
```

agliga commented on src/components/ebay-calendar/index.marko:
```diff
@@ -64,37 +69,33 @@ $ const rangeEndText = input.a11yRangeEndText ?? "end of range"
-                                                disabled=disabled
-                                                aria-label=rangeAriaLabel
+                                                disabled=isDisabled
+                                                aria-label=(a11yTexts.length > 1) && `${day}${a11yTexts.join(a11ySeparator)}`
```
Does this need a separator or something between day and a11yText's join?

LuLaValva replied:
No separator is needed, because the array starts as a single empty string so effectively it begins with a separator
